R/ivreg_marg.R

```diff
@@ -0,0 +1,100 @@
+#' Predicted Margins for `ivreg` objects from the \code{AER} package
+#'
+#' Obtains predicted margins and standard errors
+#' of those predictions from a fitted \code{ivreg} model object.
+#'
+#' @inheritParams marg
+#' @param ... additional parameters passed to \code{?marg}.
+#'
+#' @export
+#' @examples
+#' # From ?AER::ivreg
+#'
+#' # data
+#' data("CigarettesSW", package = "AER")
+#' CigarettesSW$rprice <- with(CigarettesSW, price/cpi)
+#' CigarettesSW$rincome <- with(CigarettesSW, income/population/cpi)
+#' CigarettesSW$tdiff <- with(CigarettesSW, (taxs - tax)/cpi)
+#'
+#' # model
+#' fm <- AER::ivreg(log(packs) ~ log(rprice) + log(rincome) |
+#'                    log(rincome) + tdiff + I(tax/cpi),
+#'                  data = CigarettesSW, subset = year == "1995")
+#'
+#' # Get margins for different levels of price/cpi
+#' rprice_levs <- round(quantile(CigarettesSW$rprice))
+#'
+#' marg(fm, data = subset(CigarettesSW, year == "1995"),
+#'      var_interest = 'rprice', at_var_interest = rprice_levs)
+#'
+marg.ivreg <- function(mod, var_interest,
+                       data,
+                       weights = mod$weights,
+                       ...){
+
+  .marg(mod = mod, var_interest = var_interest,
+        data = data,
+        weights = weights,
+        ...)
+}
+
+pred_se.ivreg <- function(model,
+                          deriv_func = function(x) rep.int(1, length(x)),
+                          link_func = identity,
+                          ...){
+
+  .pred_se(model = model,
+           deriv_func = function(x) rep.int(1, length(x)),
+           link_func = identity,
+           ...)
+
+}
+
+#' @importFrom stats get_all_vars
+get_data.ivreg <- function(model, data, weights){
+
+  # Store original number of rows
+  nrow_orig <- nrow(data)
+
+  # Grab only necessary variables
+  data <- data[, all.vars(model$formula)]
```

Comment:
Should this have a `, drop`? I know it's kinda silly because there has to be more than one covar, but ... maybe?